heron/api/src/scala/com/twitter/heron/streamlet/scala/converter/ScalaToJavaConverter.scala

```diff
@@ -0,0 +1,45 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.streamlet.scala.converter
+
+import com.twitter.heron.streamlet.Context
+import com.twitter.heron.streamlet.scala.Sink
+
+/**
+  * This class transforms passed User defined Scala Functions, Sources, Sinks
+  * to related Java versions
+  */
+object ScalaToJavaConverter {
+
+  def toSerializableSupplier[T](f: () => T) =
+    new com.twitter.heron.streamlet.SerializableSupplier[T] {
+      override def get(): T = f()
+    }
+
+  def toSerializableFunction[R, T](f: R => _ <: T) =
+    new com.twitter.heron.streamlet.SerializableFunction[R, T] {
+      override def apply(r: R): T = f(r)
+    }
+
+  def toJavaSink[T](sink: Sink[T]): com.twitter.heron.streamlet.Sink[T] = {
```

Comment:
It seems we are overriding the java Sink and plugging in our sink, however what happens if the sink is sufficiently complicated inside those methods, do we need to account for that

Reply:
`Sink` exposes the following functions(`setup, put and cleanup`) as _public_ and other logics which are covered internally are managed and exposed to low level Topology API by these functions. (e.g: `com.twitter.heron.streamlet.impl.sinks.ComplexSink` just calls its `setup` and `put` functions. This just provides wrapper.